sql/stmtdiagnostics: TestDiagnosticsRequest failed

sql/stmtdiagnostics.TestDiagnosticsRequest [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/15835799?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/15835799?buildTab=artifacts#/) on release-23.2 @ [88018946832e96e689c8f180f9f4621ae2f61834](https://github.com/cockroachdb/cockroach/commits/88018946832e96e689c8f180f9f4621ae2f61834):


```
=== RUN   TestDiagnosticsRequest
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/cb4e70d2cc4a901d2a7689654ffd0100/logTestDiagnosticsRequest2523976549
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:159: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    statement_diagnostics_test.go:541: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/cb4e70d2cc4a901d2a7689654ffd0100/logTestDiagnosticsRequest2523976549
--- FAIL: TestDiagnosticsRequest (117.14s)
=== RUN   TestDiagnosticsRequest/plan-gist_matching
    sql_runner.go:87: error executing query="INSERT INTO large SELECT 2 FROM generate_series(1, 10000);" args=[]: dial tcp 127.0.0.1:34787: connect: connection refused
    --- FAIL: TestDiagnosticsRequest/plan-gist_matching (102.24s)
```

Parameters:
 - <code>TAGS=bazel,gss</code>
 - <code>stress=true</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDiagnosticsRequest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-39830

--- a/pkg/sql/stmtdiagnostics/statement_diagnostics_test.go
+++ b/pkg/sql/stmtdiagnostics/statement_diagnostics_test.go
@@ -46,7 +46,7 @@ func TestDiagnosticsRequest(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	skip.UnderShort(t)
-	skip.UnderDeadlock(t, "the test is too slow")
+	skip.UnderDuress(t, "the test is too slow")
 
 	srv, db, _ := serverutils.StartServer(t, base.TestServerArgs{})
 	ctx := context.Background()

